app/test_split_integer.py

```diff
@@ -1,21 +1,104 @@
+import pytest
 from app.split_integer import split_integer
 
 
-def test_sum_of_the_parts_should_be_equal_to_value() -> None:
-    pass
+@pytest.mark.parametrize(
+    "value, number_of_parts, result",
+    [
+        (12, 1, 12),
+        (12, 2, 12),
+        (-12, 5, -12),
+        (12345, 2, 12345),
+        (12345, 6, 12345),
+        (12345, 100, 12345)
+    ]
+)
+def test_sum_of_the_parts_should_be_equal_to_value(
+        value: int,
+        number_of_parts: int,
+        result: int
+) -> None:
+    output = sum(split_integer(value, number_of_parts))
+    assert (output == result), \
+        f"Sum of parts ({output}) not equal to value ({result})"
 
 
-def test_should_split_into_equal_parts_when_value_divisible_by_parts() -> None:
-    pass
+@pytest.mark.parametrize(
+    "value, number_of_parts, result",
+    [
+        (12, 2, True),
+        (12, 6, True),
+        (-12340, 10, True),
+        (12300, 100, True),
+        (12, 5, False),
+        (12340, 11, False),
```

Comment:
The expected results for these test cases should reflect whether the `split_integer` function can split the value into equal parts when divisible by the number of parts. Ensure that the expected results align with the task description.